egg/platform_ppc.py

```diff
@@ -248,17 +310,57 @@ def get_additional_include(func, platform, simd_ext):
 
     return ret
 
+
 # -----------------------------------------------------------------------------
 # Get SoA types
 
 def get_soa_typ(simd_ext, typ, deg):
-    ntyp = get_type(simd_ext, typ) if typ != 'f16' else 'float16x8_t'
+    ntyp = get_type("", simd_ext, typ, fmtspec["styp"]) if typ != 'f16' else 'float16x8_t'
     return '{}x{}_t'.format(ntyp[:-2], deg)
 
+
 # -----------------------------------------------------------------------------
 
+
+def printf2(func):
+    """
+    debugging purposes
+    juste decorate the function with it and when executed on test, it will print the environnements
+    """
+    import inspect
+
+    def wrapper(*args, **kwargs):
+        func_args = inspect.signature(func).bind(*args, **kwargs).arguments
+        func_args_str = f"{func.__name__} called on {fmtspec['typ']}\\n" + ", ".join(
+            "{} = {!r}".format(*item) for item in func_args.items())
+        ret = func(*args)
+        if not DEBUG:
+            return ret
+        if func.__name__ == "store1234" and fmtspec["typ"][1:] == "64":
+            ret += """
+                   int k;
+                   printf("element to store: %llx %llx", {in1}.v0, {in1}.v1);
```

Comment:
`%llx` is not standard C89 nor C++98 I think. For Unixes `%ld` should work because `sizeof(long) == 8`. On Windows `%llx` shoudl be used according to the documentation.